File: Go/Leetcode1.go
import "fmt"

func twoSum(nums []int, target int) []int {
    compliments := make(map[int]int)

    for index, element := range nums {
        compliment := target - element
        if index1, present := compliments[compliment]; present {
            return []int{index1, index}
        }else {
            compliments[element] = index
        }
    }
    return []int{}
}

File: Java/Leetcode1.java
import java.util.Hashtable;

class Leetcode1 {
    public int[] twoSum(int[] nums, int target) {
        Hashtable<Integer, Integer> compliments = new Hashtable<Integer, Integer>();
        for(int i = 0; i < nums.length; i++) {
            int compliment = target - nums[i];
            if(compliments.containsKey(nums[i])) {
                return new int[] {compliments.get(nums[i]), i};
            }else {
                compliments.put(compliment, i);
            }
        }
        return null;
    }

}
